{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Acquiring pathways based on genes associated with inflammation.\n",
    "To start off, we would like to obtain genes that are associated with inflammation. For this purpose we will acquire these genes from one databases.\n",
    "The database we will use is DisGeNET. This database can be found at [DisGeNET](http://www.disgenet.org/).\n",
    "\n",
    "The genesets have to be downloaded manually. For DisGeNET the following link has to be used: [DisGeNET inflammation genes](http://www.disgenet.org/browser/0/1/0/C0021368/),\n",
    "The search query used was \"Inflammation\".\n",
    "\n",
    "The databases has its own approach of scoring the association of these genes with inflammation. The scoring methodology is described in detail here: [DisGeNET](http://www.disgenet.org/dbinfo). We opted a score of at least 0.01 in this case. \n",
    "The criteria for pathway selection can be set down below.\n",
    "\n",
    "In this notebook we will select pathways from 3 databases, i.e. [KEGG](https://www.genome.jp/kegg/), [REACTOME](https://reactome.org/) and [WikiPathways](https://www.wikipathways.org/index.php/WikiPathways).\n",
    "\n",
    "#### Pathway selection criteria variables:\n",
    "\n",
    "n = number of genes a pathway has to contain at least\n",
    "\n",
    "k = number of genes associated with inflammation a pathway has to contain at least\n",
    "\n",
    "z = percentage of genes wihtin that pathway that have to be associated with inflammation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'C:/Users/Laurent/Documents/GitHub/inflammation_networks2'"
      ],
      "text/latex": [
       "'C:/Users/Laurent/Documents/GitHub/inflammation\\_networks2'"
      ],
      "text/markdown": [
       "'C:/Users/Laurent/Documents/GitHub/inflammation_networks2'"
      ],
      "text/plain": [
       "[1] \"C:/Users/Laurent/Documents/GitHub/inflammation_networks2\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# check wd\n",
    "getwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "\"package 'qusage' was built under R version 3.5.2\"\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "The following objects are masked from 'package:plyr':\n",
      "\n",
      "    arrange, count, desc, failwith, id, mutate, rename, summarise,\n",
      "    summarize\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "Warning message:\n",
      "\"package 'tidyr' was built under R version 3.5.2\""
     ]
    }
   ],
   "source": [
    "# load libraries\n",
    "library(limma)\n",
    "library(qusage)\n",
    "library(plyr)\n",
    "library(dplyr)\n",
    "library(tidyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>diseaseid</th><th scope=col>disease_name</th><th scope=col>geneid</th><th scope=col>symbol</th><th scope=col>uniprot</th><th scope=col>protein_class</th><th scope=col>gene_name</th><th scope=col>dsi</th><th scope=col>dpi</th><th scope=col>pli</th><th scope=col>score</th><th scope=col>el</th><th scope=col>ei</th><th scope=col>npmids</th><th scope=col>nsnps</th><th scope=col>year_initial</th><th scope=col>year_final</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>C0021368                         </td><td>Inflammation                     </td><td>3576                             </td><td>CXCL8                            </td><td>P10145                           </td><td>signaling molecule               </td><td>C-X-C motif chemokine ligand 8   </td><td>0.342                            </td><td>0.862                            </td><td>       NA                        </td><td>0.40                             </td><td>no reported evidence             </td><td>NA                               </td><td>13                               </td><td>0                                </td><td>2003                             </td><td>2015                             </td></tr>\n",
       "\t<tr><td>C0021368                         </td><td>Inflammation                     </td><td>1401                             </td><td>CRP                              </td><td>P02741                           </td><td>                                 </td><td>C-reactive protein               </td><td>0.399                            </td><td>0.862                            </td><td>0.0037357                        </td><td>0.40                             </td><td>no reported evidence             </td><td>NA                               </td><td>18                               </td><td>0                                </td><td>2001                             </td><td>2011                             </td></tr>\n",
       "\t<tr><td>C0021368                         </td><td>Inflammation                     </td><td>7124                             </td><td>TNF                              </td><td>P01375                           </td><td>signaling molecule               </td><td>tumor necrosis factor            </td><td>0.263                            </td><td>0.966                            </td><td>0.8046000                        </td><td>0.40                             </td><td>no reported evidence             </td><td>NA                               </td><td>26                               </td><td>0                                </td><td>2001                             </td><td>2014                             </td></tr>\n",
       "\t<tr><td>C0021368                         </td><td>Inflammation                     </td><td>7040                             </td><td>TGFB1                            </td><td>P01137                           </td><td>signaling molecule               </td><td>transforming growth factor beta 1</td><td>0.336                            </td><td>0.931                            </td><td>0.1718200                        </td><td>0.38                             </td><td>no reported evidence             </td><td>NA                               </td><td>10                               </td><td>0                                </td><td>2002                             </td><td>2010                             </td></tr>\n",
       "\t<tr><td>C0021368                         </td><td>Inflammation                     </td><td>3569                             </td><td>IL6                              </td><td>P05231                           </td><td>                                 </td><td>interleukin 6                    </td><td>0.287                            </td><td>0.966                            </td><td>0.3387300                        </td><td>0.37                             </td><td>no reported evidence             </td><td>NA                               </td><td>18                               </td><td>0                                </td><td>2001                             </td><td>2016                             </td></tr>\n",
       "\t<tr><td>C0021368                         </td><td>Inflammation                     </td><td>3553                             </td><td>IL1B                             </td><td>P01584                           </td><td>                                 </td><td>interleukin 1 beta               </td><td>0.312                            </td><td>0.931                            </td><td>0.1256800                        </td><td>0.37                             </td><td>no reported evidence             </td><td>NA                               </td><td>16                               </td><td>0                                </td><td>1999                             </td><td>2017                             </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       " diseaseid & disease\\_name & geneid & symbol & uniprot & protein\\_class & gene\\_name & dsi & dpi & pli & score & el & ei & npmids & nsnps & year\\_initial & year\\_final\\\\\n",
       "\\hline\n",
       "\t C0021368                          & Inflammation                      & 3576                              & CXCL8                             & P10145                            & signaling molecule                & C-X-C motif chemokine ligand 8    & 0.342                             & 0.862                             &        NA                         & 0.40                              & no reported evidence              & NA                                & 13                                & 0                                 & 2003                              & 2015                             \\\\\n",
       "\t C0021368                          & Inflammation                      & 1401                              & CRP                               & P02741                            &                                   & C-reactive protein                & 0.399                             & 0.862                             & 0.0037357                         & 0.40                              & no reported evidence              & NA                                & 18                                & 0                                 & 2001                              & 2011                             \\\\\n",
       "\t C0021368                          & Inflammation                      & 7124                              & TNF                               & P01375                            & signaling molecule                & tumor necrosis factor             & 0.263                             & 0.966                             & 0.8046000                         & 0.40                              & no reported evidence              & NA                                & 26                                & 0                                 & 2001                              & 2014                             \\\\\n",
       "\t C0021368                          & Inflammation                      & 7040                              & TGFB1                             & P01137                            & signaling molecule                & transforming growth factor beta 1 & 0.336                             & 0.931                             & 0.1718200                         & 0.38                              & no reported evidence              & NA                                & 10                                & 0                                 & 2002                              & 2010                             \\\\\n",
       "\t C0021368                          & Inflammation                      & 3569                              & IL6                               & P05231                            &                                   & interleukin 6                     & 0.287                             & 0.966                             & 0.3387300                         & 0.37                              & no reported evidence              & NA                                & 18                                & 0                                 & 2001                              & 2016                             \\\\\n",
       "\t C0021368                          & Inflammation                      & 3553                              & IL1B                              & P01584                            &                                   & interleukin 1 beta                & 0.312                             & 0.931                             & 0.1256800                         & 0.37                              & no reported evidence              & NA                                & 16                                & 0                                 & 1999                              & 2017                             \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| diseaseid | disease_name | geneid | symbol | uniprot | protein_class | gene_name | dsi | dpi | pli | score | el | ei | npmids | nsnps | year_initial | year_final |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| C0021368                          | Inflammation                      | 3576                              | CXCL8                             | P10145                            | signaling molecule                | C-X-C motif chemokine ligand 8    | 0.342                             | 0.862                             |        NA                         | 0.40                              | no reported evidence              | NA                                | 13                                | 0                                 | 2003                              | 2015                              |\n",
       "| C0021368                          | Inflammation                      | 1401                              | CRP                               | P02741                            |                                   | C-reactive protein                | 0.399                             | 0.862                             | 0.0037357                         | 0.40                              | no reported evidence              | NA                                | 18                                | 0                                 | 2001                              | 2011                              |\n",
       "| C0021368                          | Inflammation                      | 7124                              | TNF                               | P01375                            | signaling molecule                | tumor necrosis factor             | 0.263                             | 0.966                             | 0.8046000                         | 0.40                              | no reported evidence              | NA                                | 26                                | 0                                 | 2001                              | 2014                              |\n",
       "| C0021368                          | Inflammation                      | 7040                              | TGFB1                             | P01137                            | signaling molecule                | transforming growth factor beta 1 | 0.336                             | 0.931                             | 0.1718200                         | 0.38                              | no reported evidence              | NA                                | 10                                | 0                                 | 2002                              | 2010                              |\n",
       "| C0021368                          | Inflammation                      | 3569                              | IL6                               | P05231                            |                                   | interleukin 6                     | 0.287                             | 0.966                             | 0.3387300                         | 0.37                              | no reported evidence              | NA                                | 18                                | 0                                 | 2001                              | 2016                              |\n",
       "| C0021368                          | Inflammation                      | 3553                              | IL1B                              | P01584                            |                                   | interleukin 1 beta                | 0.312                             | 0.931                             | 0.1256800                         | 0.37                              | no reported evidence              | NA                                | 16                                | 0                                 | 1999                              | 2017                              |\n",
       "\n"
      ],
      "text/plain": [
       "  diseaseid disease_name geneid symbol uniprot protein_class     \n",
       "1 C0021368  Inflammation 3576   CXCL8  P10145  signaling molecule\n",
       "2 C0021368  Inflammation 1401   CRP    P02741                    \n",
       "3 C0021368  Inflammation 7124   TNF    P01375  signaling molecule\n",
       "4 C0021368  Inflammation 7040   TGFB1  P01137  signaling molecule\n",
       "5 C0021368  Inflammation 3569   IL6    P05231                    \n",
       "6 C0021368  Inflammation 3553   IL1B   P01584                    \n",
       "  gene_name                         dsi   dpi   pli       score\n",
       "1 C-X-C motif chemokine ligand 8    0.342 0.862        NA 0.40 \n",
       "2 C-reactive protein                0.399 0.862 0.0037357 0.40 \n",
       "3 tumor necrosis factor             0.263 0.966 0.8046000 0.40 \n",
       "4 transforming growth factor beta 1 0.336 0.931 0.1718200 0.38 \n",
       "5 interleukin 6                     0.287 0.966 0.3387300 0.37 \n",
       "6 interleukin 1 beta                0.312 0.931 0.1256800 0.37 \n",
       "  el                   ei npmids nsnps year_initial year_final\n",
       "1 no reported evidence NA 13     0     2003         2015      \n",
       "2 no reported evidence NA 18     0     2001         2011      \n",
       "3 no reported evidence NA 26     0     2001         2014      \n",
       "4 no reported evidence NA 10     0     2002         2010      \n",
       "5 no reported evidence NA 18     0     2001         2016      \n",
       "6 no reported evidence NA 16     0     1999         2017      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>entrezgene</th><th scope=col>hgnc_symbol</th><th scope=col>dis_score</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>3576 </td><td>CXCL8</td><td>0.40 </td></tr>\n",
       "\t<tr><td>1401 </td><td>CRP  </td><td>0.40 </td></tr>\n",
       "\t<tr><td>7124 </td><td>TNF  </td><td>0.40 </td></tr>\n",
       "\t<tr><td>7040 </td><td>TGFB1</td><td>0.38 </td></tr>\n",
       "\t<tr><td>3569 </td><td>IL6  </td><td>0.37 </td></tr>\n",
       "\t<tr><td>3553 </td><td>IL1B </td><td>0.37 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lll}\n",
       " entrezgene & hgnc\\_symbol & dis\\_score\\\\\n",
       "\\hline\n",
       "\t 3576  & CXCL8 & 0.40 \\\\\n",
       "\t 1401  & CRP   & 0.40 \\\\\n",
       "\t 7124  & TNF   & 0.40 \\\\\n",
       "\t 7040  & TGFB1 & 0.38 \\\\\n",
       "\t 3569  & IL6   & 0.37 \\\\\n",
       "\t 3553  & IL1B  & 0.37 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| entrezgene | hgnc_symbol | dis_score |\n",
       "|---|---|---|\n",
       "| 3576  | CXCL8 | 0.40  |\n",
       "| 1401  | CRP   | 0.40  |\n",
       "| 7124  | TNF   | 0.40  |\n",
       "| 7040  | TGFB1 | 0.38  |\n",
       "| 3569  | IL6   | 0.37  |\n",
       "| 3553  | IL1B  | 0.37  |\n",
       "\n"
      ],
      "text/plain": [
       "  entrezgene hgnc_symbol dis_score\n",
       "1 3576       CXCL8       0.40     \n",
       "2 1401       CRP         0.40     \n",
       "3 7124       TNF         0.40     \n",
       "4 7040       TGFB1       0.38     \n",
       "5 3569       IL6         0.37     \n",
       "6 3553       IL1B        0.37     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'The number of genes is 428.'"
      ],
      "text/latex": [
       "'The number of genes is 428.'"
      ],
      "text/markdown": [
       "'The number of genes is 428.'"
      ],
      "text/plain": [
       "[1] \"The number of genes is 428.\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAAMFBMVEUAAABNTU1oaGh8fHyM\njIyampqnp6eysrK9vb3Hx8fQ0NDZ2dnh4eHp6enw8PD////QFLu4AAAACXBIWXMAABJ0AAAS\ndAHeZh94AAAd6ElEQVR4nO3djXaiOhQG0KBIrfXn/d92BIuiYotwHFPde617xwqSlJNPBaJN\nO2Cy9OwOwCsQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEGCAIIEAQQJ\nAggSBBAkCCBIEECQIIAgQQBBggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIE\nECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBPibQUrf5tXmeEfPaourR91e9/JRv631k48ipVPb\nAzu721Tz/ZKi/OxbuF/yefbo1HX148MdWxm1txa/r/Lzqu0ePt/Tz/S3g7S3au+4WumrSFeP\nurXu9aMmjMePul/XQfqxs9+PahSr66X13cdbF1t9YpBG7a2rwty9aruHL/b0M/35IKWv7zv6\nVuq94+eit0snjMdZSut7O7ubX6122a1UdTuWR5BG7a07Vr6xaruHL/b0M/3ZINX/bFez4xP1\nzZUG3z9k6SAXmxjU2fq5dVHnZ13fmvVtM6VNZ3ODnjoe7klBCni2i5ZNR+5y2oGzw/uldrB+\n1E/szVHG6cm5HoGz+um8MwSXs1QsNmfb6j6xd5tYLeoxvjo1vNq3sbh4Irxa6b7O7japfd+3\n39bs49CzbVWkojr2cr9uZ3PDgtRto97gvgfzYzsXfT7spLNW217XCd/uV9keujrr3Vtf9X7Z\n3Nh43y4+drLTrZur3t5gHrLpyF1OO3B1eI98uGNTfO/6+UWRZ+1d3w/+fh919kbrRpDat1xl\nu1J1+PksSd2VrgbKgM7WL0jV5S/ZrtD28ntA3xOkszaOP1X9v9hhJ5212vju277zzfmOz5Q+\n+vbWYb8Um/6N9+3ii05WP63at4evtvRMmXTjTqfdtz28DzrcsWhKvd3v9OV5kQ9j4DQEvxXd\nbV0Pjfre8rhyef7Y7iHu2UpX5R3Q2XqcXL3Zb0PQ9nLdffSwIJ21cdxg89rX+4t9XrTaOLwE\n7Rbfv/S8eYt5vbe6++V64327+OL37L5aX67au4evtvRMmXTjTp3d1x326XAYcTZeD69A286j\n0uG82Ko4PMOele7yx/ppeLndv0M61bl+7OKsfj0r3dvZnvGwbLq9XZx6WTZ5OPXzqKel4x2n\nNvYbLNZNqma9fW520nmrB7PmPd0pMz1xrvfL/jWsOgTw1g65eNDx9zx168aqv2wwB9l05C43\nxmb93LZYXa6ULk47tz+vjm8ULpZ2flwcnsubIbI4PXZ7VsGele7t7HGdsyfh7/i3vdw2w/Se\nIJ21UX73vTkEu/GLXbR68FGn6qtp6aveax+9e6uzX27tkN4gnXXrxqq/bDAH2XTkLjfG5vel\nmMXVceu2e0d30O5uVvm48PDYTe/Kx+30rnRHZ3uCdMpJ0a7QHK3cE6T+Nm70uX3V7rS6a1dZ\n1EP4s26+OrzI9e6tgXvtvJv9P52vOqgMz5VNR+5y2oHnb+OqdhD0PLmNDdLuh5Uve/NLkG53\ntmyP7vuCdPYqtukdjZcttbptDOpzTzzr93ZF81+x735xdmR5M0g/7bVRQfppgznIpiN3Oe3A\n1flr/fbzcHpnvuvZ25dPb8OCdHwqLPq2ufthpTs6+3ExFWLXhOZqI/vXhPKuIHXbGNTnom9o\n7uO4+n41Wp2fXbsZpJ/22qgg/V6G58qmI3c57cD59fH9anHjaetUleZA+nSMVBfpq3dolEPe\nnN9a6Y7O1gcXFwd3Zeee42Pmx9eKgUHqtDHvHozc6vN5q98Ox0er43HS7ioTv+61vl18tlNO\nx0g9qw4qw3Nl05G7tDvwqzw7hz07vtFvn7a2u94g1Umqz9rVh81F8xz7VXRKd3rUoNNFA8/a\n/djZ+rHlYWZDcXz1qU+EfZ69vK7vC9JZG7+dtWsecd5qq0jH56bvY6fevXVrr93YxQfnZ+36\nV3XW7kFSR+dAer+/55vjFcS6GNWuP0gHRV3Oxenn3fWjTlPgFp1NXD+p9q10T2fP59qVbU8a\nncuwh95ebvX8AKLjeoc0lj/0+azVVvW92iJdzPi73Mf9e+3GLt5dtLi8veqgMjxVNh25y+XQ\nbPdoe2zdPJsu0vmxUidIZeexm8Pt6nvp5aPaEp5/XOCigv0r3dPZ7uzv2WG11fePZ1f8t+0Q\nS12nli521FkbX70zGy76fNZqq36n9fm98Ku7/uXe6t9rN3Zxu/Fut26tOqgMz5RNR+7SDqB5\ntT3e0fzbHA7MD++n63fWZ4cspyG4W3bmha33j5l/do8Szh61WhS/T/LqXemuztbz4spmreNL\nQTMHrbz46MXyriCdt1FPo2s3eLPP3VaPitPZ8fP1yxsHLWcbv7GLT79np1u3Vh1UhifKpiPw\nlwkSBBAkCCBIEECQIIAgQQBBggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIE\nECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAgQQBBggCCBAEECQIIEgQQJAgg\nSBBAkCCAIEEAQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEGCAOOD9PVRplpZfQX2B/6ksUHa\nztLJPLRL8PeMDVKVis91c2uzKlIV1yH4i8YGqUjr4+11KmI6A3/V2CCldOsHeENekSDAhGOk\n1aa55RgJxp/+nnfO2s22kV2Cv2fCdaSquY5UlB+uI/H2nCaAAIIEAUwRggCmCEEAU4QggAuy\nEOBBU4RS1+1tBBv5u8Bk/+EV6XaQRrb9v7YHg/2HKUKCxOv7D1OEBInX9x+mCAkSr+8/DD5B\n4vUJEgSYOviWRZotRzYhSLyM0YNvXaZiufsYMEVIkHh9YwffuklQlRbb3aZMP74mCRKvb+zg\nW9TXjqrDldhtmo1qQpB4GdOmCKWy88PdTQgSL2NakD4P7+lMEeLdjX9rt2inM2wXpgjx7kZ/\nsK84vp9Lv3yKQpB4feMHX9XGp/jlY32CxOszswECCBIEECQIIEgQQJAggCBBAEGCAIIEAQQJ\nAggSBBAkCCBIEECQIIAgQQBBggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIE\nECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAgQQBBggCCBAEECQIIEgQQJAgg\nSBBAkCCAIEEAQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQ\nIIAgQQBBggCCBAHGD76vjzLVyuprZBOCxMsYO/i2s3QyH9eEIPEyxg6+KhWf6+bWZlWkalQT\ngsTLGDv4irQ+3l6nYlQTgsTLGDv4Urr1w/AmBImX4RUJAkw4RlptmluOkWD84Jt3ztrNtqOa\nECRexoTrSFVzHakoP1xH4u2Z2QABBAkCTBx8y1lK5WpkE4LEy5h2Hen7jMOPJ+0EiTcwKUhV\nqra73aZKy1FNCBIvY1KQitSc996m2agmBImXMSlI7dSg6ylCqSu67f+1PRhsUpAWbZBMEeLN\njQ9S+bFcpc/9zW1lihDvbnyQjm/bUipMEeLNjR586/VyWZbNKYfqxxwJEm/AzAYIIEgQYPTg\n2y5Smn9PDvIJWd7d6G8RKg7fxXXYiCDx5sZ/Qna5T9OyaL6JS5B4d+O/s6H5Z1PMNoIEU79F\naDufCxKMHXyz1F48ms0Fibc3dvAt0+L71ibNBYl3N3rwVcf0rH6Y4P1jE4LEyxg/+NZle2uz\nECTenJkNEECQIIAgQQBBggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIEECQI\nIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAgQQBBggCCBAEECQIIEgQQJAggSBBA\nkCCAIEEAQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAg\nQQBBggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIEECQIIEgQYPzg+/ooU62s\nvkY2IUi8jLGDbztLJ/NxTQgSL2Ps4KtS8blubm1WRapGNSFIvIyxg69I6+PtdSpGNSFIvIyx\ngy+lWz8Mb0KQeBlekSDAhGOk1aa55RgJxg++eees3Ww7qglB4mVMuI5UNdeRivLDdSTenpkN\nEECQIIApQhDAFCEIYIoQBHBBFgI8aIpQ6opu+39tDwbzigQBTBGCAKYIQQBThCCAmQ0QQJAg\nwOjBt12kNF99b8QnZHlzo6cIFYeJdoeNCBJvbvzp7+U+TcuimWYnSLy78Rdkm382xWwjSDB1\nitB2PhckGDv4Zqm9CDubCxJvb+zgW6bF961NmgsS72704KuO6Vn9MMH7xyYEiZcxfvCty/bW\nZiFIvDkzGyCAIEEAQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBI\nEECQIIAgQQBBggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIEECQIIEgQoDv4\nZh+bRzcxbEF0Q/Bo3cGXUnpElgSJ19cdfNvPxSOyJEi8vsvB9/Uxi86SIPH6egbfuti/Li0f\n2sQvC6Ibgke7HnyrearNH9jEbwuiG4JHuxh824/9y9Fstd2nqXxQEwMWRDcEj3Y2+L7qkw3V\n+rAgbFgKEq/v7DrS/sVouW0XFI9oYtiC6Ibg0c6uI5WrRzcxbEF0Q/BoZ9eRHt/EsAXRDcGj\nnQ2+bVW/nyuq2EQJEq+vO/g2RXOGIaUidG6DIPH6uoNvnhb1a9G2ijv1fdnEsAXRDcGjnU9a\nvbwR3sSwBdENwaN1B1+RDgdHW0GC+3QHX5XmX/t/vuapelQTwxZENwSPdjb4DrPsIufZXTUx\naEF0Q/Bo54Pvs6xjFDjz+7qJIQuiG4JH850NEECQIIAgQYCzwVd/zPzgYU0MWhDdEDxad/B9\npCRIMMb5Bdng83XXTQxbEN0QPFrvFKHHNTFsQXRD8GjdwVemh3wiSZB4fecfo2imCD2yiWEL\nohuCR7v4ymInG2AMQYIALshCAEGCAOeDb1XW7+rKQV/Z8PVRNu8Cy+qXMxSCxOu7/jxS/d2Q\nvydpO+scUf38+SVB4vV1B98yzZtPmS/T4tfHVan4PHy38WZV/PyJWkHi9V1+Z8P3F3L9+rgi\nrY+31z9/vbEg8foupwgNDVK6fOCwJoYtGEmQeJru4Jt9vyKt0+zXx3lFgo6eY6TVkFng+2Ok\n1eGUhGMkOB985R3fIjTvnLWb/TjZVZB4fdfXkVL5OeiRX1WTu6L8cB2Jt2dmAwQQJAgwcfAt\nZ+nXv/MnSLy+sR+jOKzyfcbh568KFyRe36QgVan+436b6ufT5YLE6+sZfF/zAX9nrAnS99+B\n2f58AVeQeH19g287YNLq2VSi61ewlAa8vAkSL6N38A19a7dog2SKEG+ub/Atfw7G4XGp/Fiu\nUn3tdluZIsS76z/Z8PH7405v21IqTBHizfUFaTbkm4vX6+WyLJtTDtXP3yspSLw+MxsggCBB\ngBsXZCO/JFKQeH2CBAHOBt9HUc8//SqGfLBvZBODFkQ3BI/WHXwf39/DsE6/zxG649VLkHh9\nvV8GNOBt3VKQ4OT8e+3aV6Tfv0Votx78BlCQeH3dwVd/M9D+n0HfIlTH7eePIfU2MWzBSILE\n05wNvvabgYYlZNn5arvBTQxaMJIg8TTng++z+RahXz46Pq2JIQuiG4JHM7MBAggSBLj+gsih\nf2hsZBNDFkQ3BI92fbJhN+gPjY1tYtCC6Ibg0bqD754/NDayiWELohuCRzu/IDv8D42NbGLY\nguiG4NEupwgJEozQHXz3/KGxkU0MWxDdEDxazzHSwClCo5oYtiC6IXi0s8F3zx8aG9nEoAXR\nDcGjXV9HGvqHxkY2MWRBdEPwaGY2QIDu4CsHfi5iQhPDFkQ3BI/W+wnZxzUxbEF0Q/Bol6e/\nH9zEsAXRDcGjdQfftpz/8gfKJzcxbEF0Q/BoN77X7lFNDFsQ3RA8miBBAKe/IYAgQYA7vhNy\nahPDF0Q3BI92HqSHxEmQeH2CBAEECQIIEgQQJAggSBDgFKSH/NnLbhPDF0Q3BI8mSBDAzAYI\nIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAgQQBBggCCBAEECQIIEgQQJAggSBBA\nkCCAIEEAQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAg\nQYDxg+/ro2z+bnNZfY1sQpB4GWMH33bW+Rvo83FNCBIvY+zgq1LxuW5ubVZFqkY1IUi8jLGD\nr0jr4+11KkY1IUi8jLGDL6VbPwxvQpB4GV6RIMCEY6TVprnlGAnGD75556zdbDuqCUHiZUy4\njlQ115GK8sN1JN6emQ0QQJAggClCEMAUIQhgihAEcEEWAjxoilDqim77f20PBvOKBAFMEYIA\npghBAFOEIICZDRBAkCCAIEEAQYIAggQBxs9sGDR54ccmBImXMXbwLQUJTkYPvnXx84cnBjQh\nSLyM8YNv/fPEoAFNCBIvY8LgW3bmrY5qQpB4Gc7aQQBBggCCBAEECQIIEgQQJAggSBBAkCCA\nIEEAQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAgQQBB\nggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEGCAIIE\nAQQJAggSBBAkCCBIEECQIIAgQQBBggCCBAEECQIIEq0U7dm/0P8kSLTUYwJBoqUeEwgSLfWY\nQJBoqccEgkRLPSYQJFrqMYEg0VKPCQSJlnpMIEi01GMCQaKlHhMIEi31mECQaKnHBIJESz0m\nECRa6jGBINFSjwkEiZZ6TCBItNRjAkGipR4TjP9lvz7K5vPEZfU1sgmFy4t6TDD2l93OOp/N\nn49rQuHyoh4TjP1lq1R8rptbm1WRqlFNKFxe1GOCsb9skdbH2+tUjGpC4fKiHhOM/WXPvmvp\n5y9eEqQ/Qj0m8IpESz0mmHCMtNo0txwjvQr1mGD0LzvvnLWbbUc1oXB5UY8JJlxHqprrSEX5\n4TrSa1CPCcxsoKUeEzzolx30XeoKlxf1mGDiL7ucpVSuRjahcHlRjwmmXUf6PuPw40k7Qfor\n1GOCSUGqUrXd7TZVWo5qQuHyoh4TTApSkZrz3ts0G9WEwuVFPSaYFKT2PIIpQi8hvB7v9BcA\nJwVp0QbJFKFXkHs9sq7v+CCVH8tV+tzf3FamCL2E3OuRdX3HB+n4aptSYYrQK8i9HlnXd3Tn\n1uvlsiybUw7VjzkSpL8i93pkXV9ThGjlXo+s6ytItHKvR9b1FSRaudcj6/oKEq3c65F1fQWJ\nVu71yLq+gkQr93pkXV9BopV7PbKuryDRyr0eWddXkGjlXo+s6ytItHKvR9b1FSRaudcj6/oK\nEq3c65F1fQWJVu71yLq+gkQr93pkXV9BopV7PbKuryDRyr0eWddXkGjlXo+s6ytItHKvR9b1\nFSRaudcj6/oKEq3c65F1fQWJVu71yLq+gkQr93pkXV9BopV7PbKuryDRyr0eWddXkGjlXo+s\n6ytItHKvR9b1FSRaudcj6/oKEq3c65F1fQWJVu71yLq+gkQr93pkXV9BopV7PbKuryDRyr0e\nWddXkGjlXo+s6ytItHKvR9b1FSRaudcj6/oKEq3c65F1fQWJVu71yLq+gkQr93pkXV9BopV7\nPbKuryDRyr0eWddXkGjlXo+s6ytItHKvR9b1FSRaudcj6/oKEq3c65F1fQWJVu71yLq+gkQr\n93pkXV9BopV7PbKuryDRyr0eWddXkGjlXo+s6ytItHKvR9b1FSRaudcj6/oKEq3c65F1fQWJ\nVu71yLq+gkQr93pkXV9BopV7PbKuryDRyr0eWddXkGjlXo+s6ytItHKvR9b1FSRaudcj6/oK\nEq3c65F1fQWJVu71yLq+gkQr93pkXV9BopV7PVK00M5FbuzOJnIv3LvJvR5Zb0+QaOVej6y3\nJ0i0cq9H1tsTJFq51yPr7QkSrdzrkfX2BIlW7vXIenuCRCv3emS9PUGilXs9st6eINHKvR5Z\nb0+QaOVej6y3J0i0cq9H1tsTJFq51yPr7QkSrdzrkfX2BIlW7vXIenuCRCv3emS9PUGilXs9\nst7eSwUp5w9+/QFZD9Tct/dSQcp8e7nLff9lvb3xG/v6KJun7bL6GtlE1jvmAdvLXe77L+vt\njd3YdtZ5CzQf10TWO+YB28td7vsv6+2N3ViVis91c2uzKlI1qomsd8wjtpf5MZztPWFjRVof\nb69TMaqJrHeM7dne/9jY2dPh9XPjoCfO8GdouMfIsd8/mEc+7o5XJHh9E46RVpvm1q/HSPD6\nRr+8zTsvkbNtZJfg75lwHalqriMV5ccv15Hg9b3btRJ4CEGCAIIEAQQJAggSBBAkCCBIEECQ\nIIAgQQBBggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAzwzSk76ECQ5CB3Pkxv5Q20Po3zRv\n1T9Buk3/pnmr/gnSbfo3zVv1T5Bu079p3qp/gnSb/k3zVv0TpNv0b5q36p8g3aZ/07xV/wTp\nNv2b5q36J0i36d80b9U/QbpN/6Z5q/4J0m36N81b9U+QbtO/ad6qf7n/svAnCBIEECQIIEgQ\nQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAgQQBBggCCBAH+e5CqIhXV9qc7nqunO8uc\nnm2u+recZb3/touUFuvn9edS33D7Cijw/x4j8+bPAMx+uOO5erqzjv2zBdNc9a9q7ihySdJV\n/4rmjmyS1DfctsXfC9JXKta7dZG+bt7xXD3d2f+UT5Cu+rdOi239mrl4Zq9OrvpX1T2rUvnM\nTnX0DrcyosD/eYxUabX//2f6uHnHc113Z5nmGQXpqn/loW+5dPGqf0WqXytz6V7vcPsM+UtJ\n//k3LNNmVz+NljfveK7r7qQqn2Fwe3fl0sUb/UvFU3pzrad/m5hnyv9cgHT5BHp1x3Ndd2ed\nT+d2N3fXNs2f0Jke/f2r0vIZnenR07952ghSuN7u5NK53c3dtWzesWSgr3/7t07Vc3pz7bp/\nH+kzpMCCdOZvBmlTZPLOuLd/y7LI5RD4un/NuzxBCvcng7QtMnljd7Oci1ze2131b1ZfOPiD\nQSouf5GrO56rtzu5dG53o3/zbK7C3SrnNpezDZf9WzTvif9gkA5nTTaXZ+02eZ21u+hORkHq\n6d9mNt88rT+XbpUzl1142b90NHXL//kX/GieAVano8+rO56rtzu5jIJdX/9WuZywa1z173Ad\naZPL1JXL/v3ZIP3BmQ1ZBemqf5uscnRjZsO2zOUYqX+4/cG3drtZE/+m+Ifud+7IwVX/dlkF\n6ap/i6hn1CBX+6/Iv75/MkjbZvbtoel0cUcOrvq3yytIl/0Le2sS5Hr/7e+YZfJ6tOuv758M\nErwkQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAgQQBB\nggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIEECQIIEgQQJAggCBBAEH6A7bV\nLKV5Pn/2jmuClL/t4e+wpmL77J5wkyDlb5Hmm91uM0/5/KldLglS/lJqXoq22fzFZa6pTf7O\nA1QVzQvU3nL2/ffC90mbpfJwT+FQ6ikEKX9VWmyOP8yPR0vNrTTf1UEqU/2+rzzew/8mSH/A\nPjKz6qu5+Znm2/1BU1XfKta7dZE+6yDN62St6n+287R6bm/fkyD9BatF/TJUB6RMX/XRUlHf\nqn9e1S9AKTUpK5tjqW3zJo//TJD+iK+Poo7L6Xjp+1b9z/Hmt+f08L3Z6X/GOs0EKVt2evaO\nwehkZtcfpP/eN1r2ffbKdDij3RwZza+OkcpjgkqnGZ5HkLL3ldJyu/9nXgdqWZ+Zq67O2jUr\nNvfs13Cy4QkEKX/V96FPc4Go/zrSYcXDPcXm9qZ4FEH6A9aLYp+Yz8MP+1iV3zMbiuPMhu8V\nl7PUvXjL/yNIEECQIIAgQQBBggCCBAEECQIIEgQQJAggSBBAkCCAIEEAQYIAggQBBAkCCBIE\nECQIIEgQQJAggCBBAEGCAIIEAQQJAggSBBAkCCBIEECQIIAgQQBBggCCBAEECQIIEgT4Bxs9\nTTG+PxeFAAAAAElFTkSuQmCC",
      "text/plain": [
       "Plot with title \"Distribution of DisGeNET score without cutoff\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# load in disgenet file\n",
    "disgenet <- read.delim(file.path(getwd(), \"data-input\", \"C0021368_disease_gda_summary.tsv\"))\n",
    "\n",
    "# check head of disgenet file\n",
    "head(disgenet)\n",
    "disgenet <- disgenet[,c(\"geneid\", \"symbol\", \"score\")]\n",
    "colnames(disgenet)[c(1,2,3)] <- c(\"entrezgene\", \"hgnc_symbol\", \"dis_score\")\n",
    "hist(disgenet$dis_score, xlab = \"Score\", main = \"Distribution of DisGeNET score without cutoff\")\n",
    "\n",
    "# filtered\n",
    "infl <- disgenet[disgenet$dis_score > 0.01,]\n",
    "head(infl)\n",
    "paste0(\"The number of genes is \", nrow(disgenet), \".\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pathway databases and pathway selection\n",
    "We have loaded in our DisGeNET geneset, and filtered the genes based on a score cutoff of > 0.01. Now, we will load in the .gmt files from the KEGG, Reactome and WikiPathways databases. Next we will set the pathway selection criteria and select pathways from these three databases based on our DisGeNET genes associated with inflammation. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in readLines(file):\n",
      "\"incomplete final line found on 'C:/Users/Laurent/Documents/GitHub/inflammation_networks2/data-input/wikipathways-20190410-gmt-Homo_sapiens.gmt'\""
     ]
    }
   ],
   "source": [
    "# KEGG\n",
    "kegg <- read.gmt(file.path(getwd(), \"data-input\", \"c2.cp.kegg.v6.2.entrez.gmt\"))\n",
    "kegg <- ldply(kegg, data.frame)\n",
    "colnames(kegg)[c(1,2)] <- c(\"pathway\", \"entrezgene\")\n",
    "\n",
    "### REACTOME\n",
    "reactome <- read.gmt(file.path(getwd(), \"data-input\", \"c2.cp.reactome.v6.2.entrez.gmt\"))\n",
    "reactome <- ldply(reactome, data.frame)\n",
    "colnames(reactome)[c(1,2)] <- c(\"pathway\", \"entrezgene\")\n",
    "\n",
    "### WikiPathways\n",
    "wp <- read.gmt(file.path(getwd(), \"data-input\", \"wikipathways-20190410-gmt-Homo_sapiens.gmt\"))\n",
    "wp <- ldply(wp, data.frame)\n",
    "colnames(wp)[c(1,2)] <- c(\"pathway\", \"entrezgene\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>pathway</th><th scope=col>entrezgene</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>KEGG_GLYCOLYSIS_GLUCONEOGENESIS</td><td>55902                          </td></tr>\n",
       "\t<tr><td>KEGG_GLYCOLYSIS_GLUCONEOGENESIS</td><td>2645                           </td></tr>\n",
       "\t<tr><td>KEGG_GLYCOLYSIS_GLUCONEOGENESIS</td><td>5232                           </td></tr>\n",
       "\t<tr><td>KEGG_GLYCOLYSIS_GLUCONEOGENESIS</td><td>5230                           </td></tr>\n",
       "\t<tr><td>KEGG_GLYCOLYSIS_GLUCONEOGENESIS</td><td>5162                           </td></tr>\n",
       "\t<tr><td>KEGG_GLYCOLYSIS_GLUCONEOGENESIS</td><td>5160                           </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|ll}\n",
       " pathway & entrezgene\\\\\n",
       "\\hline\n",
       "\t KEGG\\_GLYCOLYSIS\\_GLUCONEOGENESIS & 55902                              \\\\\n",
       "\t KEGG\\_GLYCOLYSIS\\_GLUCONEOGENESIS & 2645                               \\\\\n",
       "\t KEGG\\_GLYCOLYSIS\\_GLUCONEOGENESIS & 5232                               \\\\\n",
       "\t KEGG\\_GLYCOLYSIS\\_GLUCONEOGENESIS & 5230                               \\\\\n",
       "\t KEGG\\_GLYCOLYSIS\\_GLUCONEOGENESIS & 5162                               \\\\\n",
       "\t KEGG\\_GLYCOLYSIS\\_GLUCONEOGENESIS & 5160                               \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| pathway | entrezgene |\n",
       "|---|---|\n",
       "| KEGG_GLYCOLYSIS_GLUCONEOGENESIS | 55902                           |\n",
       "| KEGG_GLYCOLYSIS_GLUCONEOGENESIS | 2645                            |\n",
       "| KEGG_GLYCOLYSIS_GLUCONEOGENESIS | 5232                            |\n",
       "| KEGG_GLYCOLYSIS_GLUCONEOGENESIS | 5230                            |\n",
       "| KEGG_GLYCOLYSIS_GLUCONEOGENESIS | 5162                            |\n",
       "| KEGG_GLYCOLYSIS_GLUCONEOGENESIS | 5160                            |\n",
       "\n"
      ],
      "text/plain": [
       "  pathway                         entrezgene\n",
       "1 KEGG_GLYCOLYSIS_GLUCONEOGENESIS 55902     \n",
       "2 KEGG_GLYCOLYSIS_GLUCONEOGENESIS 2645      \n",
       "3 KEGG_GLYCOLYSIS_GLUCONEOGENESIS 5232      \n",
       "4 KEGG_GLYCOLYSIS_GLUCONEOGENESIS 5230      \n",
       "5 KEGG_GLYCOLYSIS_GLUCONEOGENESIS 5162      \n",
       "6 KEGG_GLYCOLYSIS_GLUCONEOGENESIS 5160      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# combine databases\n",
    "allDatabases <- rbind(kegg, reactome, wp)\n",
    "head(allDatabases)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select the criteria. n is the number of genes a pathway at least has to contain, k is the number of inflammation associated genes a pathway at least has to contain, z is the percentage of genes within a pathway that have to be associated with inflammation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# set criteria\n",
    "n = 10\n",
    "k = 3\n",
    "z = 25"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# filter first that pathway has at least 10 genes, so the row with the pathwayname has to occur at least 10 times in the data frame\n",
    "allDatabases <- allDatabases[allDatabases$pathway %in% names(table(allDatabases$pathway))[table(allDatabases$pathway) >= n],]\n",
    "# freq of every pathway in database\n",
    "freqallDatabases <- plyr::count(allDatabases$pathway)\n",
    "# pathway names of the rows (entrezgene IDs) that occur in the gene list associated with inflammation\n",
    "allDatabasespathways <- as.data.frame(allDatabases$pathway[allDatabases$entrezgene %in% infl$entrezgene])\n",
    "# if these pathwaynames occur at least 5 times we can create list of pathways with at least 5 inflammation genes \n",
    "# (and which were already filtered on at least 10 gesn in total)\n",
    "colnames(allDatabasespathways)[1] <- \"pathway\"\n",
    "allDatabasespathways <- as.data.frame(allDatabasespathways[allDatabasespathways$pathway %in% names(table(allDatabasespathways$pathway))[table(allDatabasespathways$pathway) >= k],])\n",
    "colnames(allDatabasespathways)[1] <- \"pathway\"\n",
    "# freq of every pathway in allDatabasespathways\n",
    "freqallDatabasespathways <- plyr::count(allDatabasespathways$pathway)\n",
    "# freqallDatabases has total amount of genes in the pathway and freqallDatabasespathways has total amount of infalmmation genes in pathway\n",
    "# combine the two data frames, so we have the total amount of genes per pathway which has at least 5 inflammation genes in them\n",
    "# and so we also have the amount of inflammation genes per pathway, so we can calculate the percentage, which has to be at least 30%\n",
    "allDatabasespathways <- as.data.frame(merge(freqallDatabasespathways, freqallDatabases, by = \"x\"))\n",
    "colnames(allDatabasespathways)[c(1,2,3)] <- c(\"pathway\", \"freqInflGene\", \"freqGene\")\n",
    "# calculate percentage and add column\n",
    "# select rows which have percentage of at least 30%\n",
    "allDatabasespathways$percentage <- ((allDatabasespathways$freqInflGene/allDatabasespathways$freqGene) * 100)\n",
    "allDatabasespathways <- allDatabasespathways[allDatabasespathways$percentage >= z, ]\n",
    "allDatabasespathways$percentage <- round(allDatabasespathways$percentage, 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# combine the retrieved pathways so we have a list of the pathways that met the criteria\n",
    "pathwayStats <- as.data.frame(allDatabasespathways)\n",
    "\n",
    "# save pathwaysStats\n",
    "write.table(pathwayStats, file.path(getwd(), \"data-output\", \"pathwayStats.txt\"), col.names = T, row.names = F, sep = \"\\t\", quote = F)\n",
    "\n",
    "# get list of only selected pathways without stats\n",
    "selectedpws <- pathwayStats[c(-2,-3,-4)]  \n",
    "\n",
    "# save selectedpws\n",
    "write.table(selectedpws, file.path(getwd(), \"data-output\", \"selected_pws.txt\"), col.names = T, row.names = F, sep = \"\\t\", quote = F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# in how many pathways are the inflammation genes\n",
    "nGenesTotal <- as.data.frame(allDatabases$entrezgene[allDatabases$entrezgene %in% infl$entrezgene])\n",
    "colnames(nGenesTotal)[1] <- \"entrezgene\"\n",
    "\n",
    "# in how many slected pathways are the inflammation genes\n",
    "selectedallDatabases <- as.data.frame(allDatabases[allDatabases$pathway %in% allDatabasespathways$pathway,])\n",
    "nInflGenesTotal <- as.data.frame(selectedallDatabases$entrezgene[selectedallDatabases$entrezgene %in% infl$entrezgene])\n",
    "colnames(nInflGenesTotal)[1] <- \"entrezgene\"\n",
    "\n",
    "# combine all frequencies together\n",
    "# first the inflammation genes in all pathways\n",
    "nGenesTotalF <- plyr::count(nGenesTotal)\n",
    "colnames(nGenesTotalF)[2] <- \"number of pathways\"\n",
    "# next the inflammation genes in selected pathways\n",
    "nInflGenesTotalF <- plyr::count(nInflGenesTotal)\n",
    "colnames(nInflGenesTotalF)[2] <- \"number of selected pathways\"\n",
    "\n",
    "# save tables\n",
    "write.table(nGenesTotalF, file.path(getwd(), \"data-output\", \"nGenes.txt\"), col.names = T, row.names = F, sep = \"\\t\", quote = F)\n",
    "write.table(nInflGenesTotalF, file.path(getwd(), \"data-output\", \"nInflGenes.txt\"), col.names = T, row.names = F, sep = \"\\t\", quote = F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.5.1 (2018-07-02)\n",
       "Platform: x86_64-w64-mingw32/x64 (64-bit)\n",
       "Running under: Windows 10 x64 (build 17134)\n",
       "\n",
       "Matrix products: default\n",
       "\n",
       "locale:\n",
       "[1] LC_COLLATE=Dutch_Netherlands.1252  LC_CTYPE=Dutch_Netherlands.1252   \n",
       "[3] LC_MONETARY=Dutch_Netherlands.1252 LC_NUMERIC=C                      \n",
       "[5] LC_TIME=Dutch_Netherlands.1252    \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] tidyr_0.8.2          dplyr_0.7.8          plyr_1.8.4          \n",
       "[4] qusage_2.16.1        limma_3.38.3         RevoUtils_11.0.1    \n",
       "[7] RevoUtilsMath_11.0.0\n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] Rcpp_1.0.0          compiler_3.5.1      pillar_1.3.1       \n",
       " [4] bindr_0.1.1         base64enc_0.1-3     tools_3.5.1        \n",
       " [7] digest_0.6.18       uuid_0.1-2          jsonlite_1.6       \n",
       "[10] evaluate_0.12       tibble_2.0.1        nlme_3.1-137       \n",
       "[13] lattice_0.20-35     pkgconfig_2.0.2     rlang_0.3.1        \n",
       "[16] Matrix_1.2-14       IRdisplay_0.7.0     IRkernel_0.8.15    \n",
       "[19] parallel_3.5.1      mvtnorm_1.0-8       bindrcpp_0.2.2     \n",
       "[22] repr_0.19.1         lsmeans_2.30-0      grid_3.5.1         \n",
       "[25] tidyselect_0.2.5    glue_1.3.0          Biobase_2.42.0     \n",
       "[28] R6_2.3.0            pbdZMQ_0.3-3        purrr_0.3.0        \n",
       "[31] magrittr_1.5        emmeans_1.3.2       htmltools_0.3.6    \n",
       "[34] BiocGenerics_0.28.0 assertthat_0.2.1    xtable_1.8-3       \n",
       "[37] estimability_1.3    crayon_1.3.4       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# information about the session.\n",
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
